lib/src/api/opt/endpoint/mod.rs

```diff
@@ -92,3 +102,49 @@ mod tests {
 		}
 	}
 }
+
+#[derive(Debug)]
+pub enum EndpointKind {
+	Http,
+	Https,
+	Ws,
+	Wss,
+	FoundationDb,
+	#[cfg(target_arch = "wasm32")]
+	IndxDb,
+	Memory,
+	RocksDb,
+	File,
+	SpeeDb,
+	TiKv,
+	Unsupported(String),
+}
+
+impl From<&str> for EndpointKind {
+	fn from(s: &str) -> Self {
+		match s {
+			"http" => Self::Http,
+			"https" => Self::Https,
+			"ws" => Self::Ws,
+			"wss" => Self::Wss,
+			"fdb" => Self::FoundationDb,
+			#[cfg(target_arch = "wasm32")]
+			"indxdb" => Self::IndxDb,
+			"mem" => Self::Memory,
+			"file" => Self::File,
+			"rocksdb" => Self::RocksDb,
+			"speedb" => Self::SpeeDb,
+			"tikv" => Self::TiKv,
+			_ => Self::Unsupported(s.to_owned()),
+		}
+	}
+}
+
+impl EndpointKind {
+	pub fn is_local(&self) -> bool {
+		!matches!(
+			self,
+			EndpointKind::Http | EndpointKind::Https | EndpointKind::Ws | EndpointKind::Wss
+		)
```

Comment:
I know we have discussed this before @sgirones but this still makes me uneasy. How about re-implimenting this method as `is_remote`. We could then re-add `is_local` as

```rust
pub fn is_local(&self) -> bool {
    !self.is_remote()
}
```

if we like.

Reply:
I agree that it looks better, changed in [f49e028](https://github.com/surrealdb/surrealdb/pull/2985/commits/f49e0287c5fe64c46a731de1f264e0768c6823f1).